sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -251,7 +259,12 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
 }
 
 @ExpressionDescription(
-  usage = "a _FUNC_ b - Logical AND.")
+  usage = "expr1 _FUNC_ expr2 - Logical AND.",
+  extended = """
+    Arguments:
+      expr1 - an expression of any type.
+      expr2 - an expression of any type.
```

Comment:
Are you sure it can support any type?

For logical operations (AND, OR or others), I think the only acceptable types are boolean